external/fv3fit/fv3fit/data/tfdataset.py

```diff
@@ -0,0 +1,43 @@
+import dataclasses
+from typing import Mapping, Sequence, Optional
+import tensorflow as tf
+from .base import TFDatasetLoader
+
+
+@dataclasses.dataclass
+class VariableConfig:
+    unstacked_dims: Sequence[str]
+
+
+@dataclasses.dataclass
```

Comment:
I have a PR where I'm fleshing it out right now, I'd like to merge it in this PR if that's OK but the implementation should be complete by the end of day today.

I'm nervous about introducing an isomorphism pattern without introducing at least one alternative registered in the code, because this has sometimes led to breaking the isomorphism pattern because nothing enforces it (particularly in DSL testing code). Having a second class, even one that's not implemented, should help make it so the code breaks more spectacularly if the isomorphism is broken.

Reply:
I'll be opening up the PR that implements this shortly.